alternator/streams.cc

```diff
@@ -77,11 +77,7 @@ struct from_string_helper {
     static ValueType& Set(ValueType& v, T data, typename ValueType::AllocatorType& a) {
         std::ostringstream ss;
         ss << data;
-#if __cplusplus >= 202002L
```

Comment:
@elcallio can you please review this change to your code? Was the usage of view() good for performance?
ostringstream::view() was added in C++20, so probably the #if is somehow wrong, or clang implemented the standard wrong. I wonder if we could have some other different ifdef that will allow us to use the new view() when it's available.

Reply:
Docs only says c++20. The gcc headers use the above version guard as to whether declare it or not. I could not find any official version guard for the feature. The usage was only an optimization, yes (and to ease the pain I feel every day that not more parts of the standard lib uses views).

Reply:
So my only remaining concern with this series is this. @elcallio do you think view() vs. str() has any performance implications? I think both of them may need to copy the string anyway (to make it contiguous) so it doesn't really matter which one we use?

Reply:
View does not need to copy anything. It just needs the original string (in the ostringstream) to live across its usage (which this does) - in fact since view is just a view, it prohibits actual data ownership. And method being const it pretty much implies no scattered buffers in iostreams. 

It is a minor perf diff. We will add a N byte copy, and probably extra malloc (since most objects serialized through here are > 32 chars), but it is hardly a deal breaker. Just a nice feature to have.

Reply:
I'm not sure that the original ostringstream is actually guaranteed to contain a **contiguous** buffer, and if it doesn't, view() needs to copy as well. But it's indeed possible that it doesn't.
So, @elcallio if you're comfortable with Avi's patch which removes this C++20 optimization, I can commit this series.
If you're answer is no - you're not comfortable with it - I wonder what other trick (constraints? SFINAE? some other ifdef?) we could use to make this C++20 method work in gcc but not in clang which for some unknown (to me) reason doesn't implement it.

Reply:
I'm fine with it.